### PR TITLE
[Bug]: Fix lazy loaded Block element in objectbrick

### LINKED ISSUE #7818: Block element data doesn't get loaded when in objectbrick and lazy loading is on.
Tested with latest version of pimcore/demo -> Version: v6.8.6

2020-12-28 17_51_57-demopimcore __ Pimcore

Steps to reproduce
-> Create block element within objectbrick and switch on lazy loading. Create a object which includes the objectbrick, add an item to the block element. -> Reload 

Expected behavior
-> Item of the block element would appear with a small delay
2020-12-28 17_53_39-demopimcore __ Pimcore
(Screenshot was taken with lazy loding switched off!)

Actual behavior
-> Block element never shows it elements
2020-12-28 17_53_04-demopimcore __ Pimcore

### PR DESCRIPTION
**Pull request by kingjia90**
## Changes in this pull request  
Resolves https://github.com/pimcore/pimcore/issues/7818

## Additional info
backport to LTS?

### CODE REVIEW

File: models/DataObject/ClassDefinition/Data/Block.php [hunk 1]
```diff
@@ -798,24 +798,30 @@ public function preGetData(mixed $container, array $params = []): mixed
...
 
+        if ($this->getLazyLoading() && !$container->isLazyKeyLoaded($this->getName())) {
+            $data = $this->load($container, $params);
+
+            $setter = 'set' . ucfirst($this->getName());
+            if (method_exists($container, $setter)) {
+                $container->$setter($data);
+            }
+        }
+        $this->preSetData($container, $data, $params);
```
Comment (fashxp) [inline models/DataObject/ClassDefinition/Data/Block.php:823]: that now gets executed for every container type, not just for concretes. Is that  a problem?

Comment (fashxp) [reply models/DataObject/ClassDefinition/Data/Block.php:823]: `$this->markLazyloadedFieldAsLoaded($container);` is missing too

**Review by fashxp — CHANGES_REQUESTED** [review]

File: models/DataObject/ClassDefinition/Data/Block.php (hunk 1 above)
Comment (kingjia90) [reply models/DataObject/ClassDefinition/Data/Block.php:823]: The `markLazyloadedFieldAsLoaded` is done within `preSetData`

https://github.com/pimcore/pimcore/blob/ee814a43916bd80cd44082e0cef6902257be5499/models/DataObject/ClassDefinition/Data/Block.php#L705-L707

and it felt like it's scanning through `localizedfields` which should't hurt for the other container types(?)
https://github.com/pimcore/pimcore/blob/ee814a43916bd80cd44082e0cef6902257be5499/models/DataObject/ClassDefinition/Data/Block.php#L709-L710

**Review by fashxp — APPROVED** [review]

**Comment by fashxp** [comment]
looks good to me :+1:

**Pull request merged**